File: deal/src/main/java/ru/artemev/deal/model/Passport.java
package ru.artemev.deal.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Past;
import javax.validation.constraints.Pattern;
import java.time.LocalDate;

@Data
@Schema(description = "Паспорт")
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Passport {

  @NotNull
  @Pattern(regexp = "[\\d]{4}", message = "Не верно введена серия паспорта")
  @Schema(description = "Серия паспорта", example = "1234")
  private String series;

  @NotNull
  @Pattern(regexp = "[\\d]{6}", message = "Не верно введена серия паспорта")
  @Schema(description = "Серия паспорта", example = "1234")
  private String number;

  @Past(message = "Дата может быть только прошлым")
  @Schema(description = "Дата выдачи")
  private LocalDate issueDate;

  @Schema(description = "Отделение")
  private String issueBranch;
}

File: deal/src/main/kotlin/ru/kit/deal/dto/Passport.kt
package ru.kit.deal.dto

import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDate
import javax.validation.constraints.NotNull
import javax.validation.constraints.Past
import javax.validation.constraints.Pattern

@Schema(description = "Паспорт")
data class Passport(
    @NotNull
    @Pattern(regexp = "[\\d]{4}", message = "Не верно введена серия паспорта")
    @Schema(description = "Серия паспорта", example = "1234")
    val series: String? = null,

    @NotNull
    @Pattern(regexp = "[\\d]{6}", message = "Не верно введена серия паспорта")
    @Schema(description = "Серия паспорта", example = "1234")
    val number: String? = null,


    @Past(message = "Дата может быть только прошлым")
    @Schema(description = "Дата выдачи")
    val issue_date: LocalDate? = null,


    @Schema(description = "Отделение")
    val issue_branch: String? = null,
)